### ISSUE TITLE
Exception when using has_and_belongs_to_many

### ISSUE THREAD

**Issue by paneq** [chilloutio/chillout#5]
```ruby
    class Entity
      has_and_belongs_to_many :followers, ->{ distinct },
        class_name: "Entity",
        join_table: "entities_following_entities",
        foreign_key: "follower_id",
        association_foreign_key: "target_id"
    end
```

```
(master) > bundle exec rails c production
Loading production environment (Rails 5.0.3)
2.3.4 :001 > e = Entity.create!(name: Time.current.to_s, state: "parked")
[Chillout] Model created: Entity
 => #<Entity id: 1, name: "2017-07-12 19:07:21 UTC", state: "parked", seatbelt: nil, created_at: "2017-07-12 19:07:21", updated_at: "2017-07-12 19:07:21"> 
2.3.4 :002 > e.follower_ids=[1]
[Chillout] Model created: 
NoMethodError: undefined method `to_sym' for nil:NilClass
Did you mean?  to_s
	from /home/rupert/develop/arkency/chillout_gem/lib/chillout/creations_container.rb:18:in `key'
	from /home/rupert/develop/arkency/chillout_gem/lib/chillout/creations_container.rb:8:in `increment!'
	from /home/rupert/develop/arkency/chillout_gem/lib/chillout/creation_listener.rb:17:in `block in listen'
	from /home/rupert/.rvm/gems/ruby-2.3.4/gems/activesupport-5.0.3/lib/active_support/callbacks.rb:396:in `instance_exec'
	from /home/rupert/.rvm/gems/ruby-2.3.4/gems/activesupport-5.0.3/lib/active_support/callbacks.rb:396:in `block in make_lambda'
	from /home/rupert/.rvm/gems/ruby-2.3.4/gems/activesupport-5.0.3/lib/active_support/callbacks.rb:232:in `block in conditional'
	from /home/rupert/.rvm/gems/ruby-2.3.4/gems/activesupport-5.0.3/lib/active_support/callbacks.rb:456:in `block in call'
	from /home/rupert/.rvm/gems/ruby-2.3.4/gems/activesupport-5.0.3/lib/active_support/callbacks.rb:456:in `each'
	from /home/rupert/.rvm/gems/ruby-2.3.4/gems/activesupport-5.0.3/lib/active_support/callbacks.rb:456:in `call'
```

## Checklist

- [x] Hotfix
- [x] Release
- [x] Test and coldfix
- [x] Release

**Comment by paneq** [comment]
Released hotfix in `0.8.9` and `0.8.5.1`.

**Comment by paneq** [comment]
Fixed: https://github.com/chilloutio/chillout/commit/52cb005ca1e84fc6ce6dc2ba31831846ac1b872f
